src/rdkafka_msg.c

```diff
@@ -478,6 +478,9 @@ rd_kafka_produceva(rd_kafka_t *rk, const rd_kafka_vu_t *vus, size_t cnt) {
                                            rd_kafka_err2str(err));
                 goto err;
         }
+        // now 'hdrs' owned by rkm as no error occurred
+        // make NULL to avoid double free in message + on error
```

Comment:
The comment should use `/* */` style comments, and IMHO it's a little verbose. Otherwise LGTM: this looks like a real bug and the correct fix.
```suggestion
        /* 'hdrs' is now owned by 'rkm' */
```